src/main/kotlin/com/sourcegraph/cody/context/ui/EnhancedContextPanel.kt

```diff
@@ -0,0 +1,128 @@
+package com.sourcegraph.cody.context.ui
+
+import com.intellij.icons.AllIcons
+import com.intellij.ide.BrowserUtil
+import com.intellij.openapi.actionSystem.ActionToolbarPosition
+import com.intellij.openapi.actionSystem.AnActionEvent
+import com.intellij.openapi.progress.ProgressIndicator
+import com.intellij.openapi.progress.ProgressManager
+import com.intellij.openapi.progress.Task
+import com.intellij.openapi.project.Project
+import com.intellij.openapi.ui.Messages
+import com.intellij.openapi.ui.VerticalFlowLayout
+import com.intellij.ui.CheckboxTree
+import com.intellij.ui.CheckedTreeNode
+import com.intellij.ui.ColorUtil
+import com.intellij.ui.ToolbarDecorator
+import com.intellij.util.ui.UIUtil
+import com.sourcegraph.cody.agent.CodyAgentService
+import com.sourcegraph.cody.agent.CommandExecuteParams
+import com.sourcegraph.common.CodyBundle
+import com.sourcegraph.common.CodyBundle.fmt
+import java.awt.Dimension
+import java.util.concurrent.atomic.AtomicBoolean
+import javax.swing.BorderFactory
+import javax.swing.JPanel
+import javax.swing.border.TitledBorder
+
+class EnhancedContextPanel(private val project: Project) : JPanel() {
+
+  val isEnhancedContextEnabled = AtomicBoolean(true)
+
+  private val refreshButton =
+      object :
+          ToolbarDecorator.ElementActionButton(
+              CodyBundle.getString("context-panel.reindex-button-name"), AllIcons.Actions.Refresh) {
+        override fun actionPerformed(p0: AnActionEvent) {
+          CodyAgentService.applyAgentOnBackgroundThread(project) { agent ->
+            ProgressManager.getInstance()
+                .run(
+                    object :
+                        Task.Backgroundable(
+                            project, CodyBundle.getString("context-panel.in-progress"), false) {
+                      override fun run(indicator: ProgressIndicator) {
+                        try {
+                          isReindexingInProgress.set(true)
+                          val cmd = CommandExecuteParams("cody.search.index-update", emptyList())
+                          agent.server.commandExecute(cmd).get()
+                        } catch (e: Exception) {
+                          val errMsg = e.message ?: e.toString()
+                          Messages.showErrorDialog(
+                              CodyBundle.getString("context-panel.error-message").fmt(errMsg),
+                              CodyBundle.getString("context-panel.error-title"))
+                        } finally {
+                          indicator.stop()
+                          isReindexingInProgress.set(false)
+                        }
+                      }
+                    })
+          }
+        }
+
+        override fun isEnabled(): Boolean {
+          return !isReindexingInProgress.get()
+        }
+
+        override fun isDumbAware(): Boolean {
+          return true
+        }
```

Comment:
Simple expressions like this can be shortened to smth like `override fun isDumbAware() = true`.